spec/test/stored_queries/test_fulltext_search.py

```diff
@@ -0,0 +1,226 @@
+                search_attrkey='scientific_name',
+                search_text=sciname,
+                ts=_NOW if sciname in scinames_test_latest else None,
+                filter_attr_expr=[{'rank': 'species'}, {'rank': 'strain'}, {'strain': True}],
+                offset=0,
+                limit=LIMIT,
+                select=['id', 'scientific_name']
+            )
+
+    def test_stored_query_validation_fail(self):
+        for sciname in scinames_test_all:
+            with self.assertRaises(Exception) as e:
+                print('\n#e', e)
+                _fulltext_query__expect_hit(
+                    self,
+                    coll=None,
+                    search_attrkey=None,
+                    search_text=None,
+                    ts=None,
+                    filter_attr_expr=None,
+                    offset=None,
+                    limit=None,
+                    select=None,
+                )
+
+    def test_arango_error(self):
+        for sciname in scinames_test_all:
+            with self.assertRaises(ArangoServerError):
+                _fulltext_query__expect_hit(
+                    self,
+                    coll='ncbi_taxon',
+                    search_attrkey='fake_attrkey',
+                    search_text=sciname,
+                    ts=None,
+                    filter_attr_expr=None,
+                    offset=None,
+                    limit=None,
+                    select=None,
+                )
+
+    def test_no_hit(self):
+        for sciname in scinames_test_all:
+            with self.assertRaises(AssertionError):
+                _fulltext_query__expect_hit(
+                    self,
+                    coll='ncbi_taxon',
+                    search_attrkey='scientific_name',
+                    search_text=sciname[::-1],
+                    ts=None,
+                    filter_attr_expr=None,
+                    offset=None,
+                    limit=None,
+                    select=None,
+                )
+
+
+# --- Test helpers ---
+
+
+def _fulltext_query__expect_hit(
+    self,
+    coll,
+    search_attrkey,
+    search_text,
+    ts,
+    filter_attr_expr,
+    offset,
+    limit,
+    select,
+):
+    """
+    Helper to run the taxonomy_search_sci_name query and make some standard
+    assertions on the response.
+    """
+    data = {
+        '@coll': coll,
+        'search_attrkey': search_attrkey,
+        'search_text': search_text,
+        'ts': ts,
+        'filter_attr_expr': filter_attr_expr,
+        'offset': offset,
+        'limit': limit,
+        'select': select,
+    }
+    resp = requests.post(
+        _CONF["re_api_url"] + "/api/v1/query_results",
+        params={"stored_query": "fulltext_search"},
+        data=json.dumps(data),
+    )
+
+    if "error" in resp.json():
+        raise ArangoServerError(resp.text)
+
+    docs = resp.json()['results']
+    hits = [doc[search_attrkey] for doc in docs]
+    self.assertIn(search_text, hits)
```

Comment:
should you check that the search text is in each doc in the `hits` array?

Reply:
I think ideally you'd specify which results you expect the search to turn up, so you wouldn't just check that the search text is in each doc, but rather that you have all the docs returned that you would expect to see.